org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Painter.java

```diff
@@ -0,0 +1,146 @@
+package org.code.neighborhood;
+
+public class Painter {
+  private static int lastId = 0;
+  private int xLocation;
+  private int yLocation;
+  private Direction direction;
+  private int remainingPaint;
+  private Grid grid;
+  private String id;
+
+  public Painter(int x, int y, String direction, int paint) {
```

Comment:
Also check for null values here and x/y out of bounds. This is the section that will be directly used by the students, so we'll want to be pretty strict on how we handle odd inputs.